rest_framework/routers.py

```diff
@@ -277,10 +285,24 @@ def get_api_root_view(self):
         for prefix, viewset, basename in self.registry:
             api_root_dict[prefix] = list_name.format(basename=basename)
 
+        view_renderers = list(api_settings.DEFAULT_RENDERER_CLASSES)
+
+        if schema_urls and self.schema_title:
+            assert coreapi, '`coreapi` must be installed for schema support.'
+            view_renderers += [renderers.CoreJSONRenderer]
```

Comment:
Wouldn't it make more sense to expect users to include the `CoreJSONRenderer` in their default renderers, and raise an error if it is missing from the defaults?